### PR TITLE
Add jira tickets auto conversion

### PR DESCRIPTION
**Pull request by sebastian-aranda**
This PR extends the `htmlToJiraMarkdown` function to support auto conversion of jira tickets to hyperlinks, e.g. DM-41184 -> [DM-41184](https://jira.lsstcorp.org/browse/DM-41184).

### CODE REVIEW

File: love/src/Utils.js [hunk 1]
```diff
@@ -1853,6 +1853,8 @@ export function trimString(string, length = 100) {
 /**
  * Function to parse HTML, generated by react-quill, to Jira Markdown.
  * Check https://jira.atlassian.com/secure/WikiRendererHelpAction.jspa?section=all for more info
+ * This function also parse Jira tickets names that are not hyperlinks.
+ * Allowed projects are in AUTO_HYPERLINK_JIRA_PROJECTS.
  * @param {string} html html to be parsed
```
Comment (wvreeven) [inline love/src/Utils.js:1858]: Not related to your PR, but it the double `html` needed here?

File: love/src/Utils.test.js [hunk 2]
```diff
@@ -20,6 +20,14 @@ describe('htmlToJiraMarkdown', () => {
...
     expect(htmlToJiraMarkdown(input)).toEqual(expectedOutput);
   });
+
+  it('should convert Jira tickets to links', () => {
+    const input =
+      '<p>This is a string with mixed content</p><h1>This is a heading.</h1><p>This is an already formatted link <a href="https://jira.lsstcorp.org/browse/DM-41184" rel="noopener noreferrer" target="_blank">DM-41184</a>.</p><p>This is a ticket name not yet formatted DM-41184.</p>';
+    const expectedOutput =
+      'This is a string with mixed content\r\nh1. This is a heading.\r\nThis is an already formatted link [DM-41184|https://jira.lsstcorp.org/browse/DM-41184].\r\nThis is a ticket name not yet formatted [DM-41184|https://jira.lsstcorp.org/browse/DM-41184].\r\n';
+    expect(htmlToJiraMarkdown(input)).toEqual(expectedOutput);
+  });
```
Comment (wvreeven) [inline love/src/Utils.test.js:30]: Consider adding a third test case, where
```
[DM-41184|https://jira.lsstcorp.org/browse/DM-41184]
```
is already present to make sure that it remains formatted correctly.

**Review by wvreeven — APPROVED** [review]
Looks good to me. I left a few comments for your consideration.

File: love/src/Utils.js (hunk 1 above)
Comment (sebastian-aranda) [reply love/src/Utils.js:1858]: Thanks and good catch! So the first `html` is the name of the variable itself, although I will adjust the description for better readability. Here are some examples from the oficial docs: https://jsdoc.app/tags-param.html.

Comment (wvreeven) [reply love/src/Utils.js:1858]: Thanks for the explanation. Perhaps place a hyphen `-` between both, as suggested by the docs link, so it is clear which is the variable name and where the description starts?

**Pull request merged**